src/Cardano/Wallet/Network/HttpBridge.hs

```diff
@@ -187,8 +190,20 @@ mkHttpBridge mgr baseUrl network = HttpBridge
         -> IO (Either ErrNetworkUnreachable a)
     defaultHandler = \case
         Right c -> return $ Right c
+
+        -- The node has not started yet or has exited.
+        -- This could be recovered from by either waiting for the node
+        -- initialise, or restarting the node.
         Left (ConnectionError e) ->
             return $ Left $ ErrNetworkUnreachable e
+
+        -- Failure response (e.g. 404 /mainnet/tip).
+        -- This can be recovered from by retrying the request.
+        Left e@(FailureResponse _) ->
+            return $ Left $ ErrNetworkUnreachable $ T.pack $ displayException e
```

Comment:
This shouldn't be in the default handler because the semantic we give to `404` is different depending on the endpoint. Actually, 404 are already handled for epochs for instance. If this is needed for the `/networkTip` endpoint, let's add it to the `getNetworkTip` handler, like done for [getEpoch](https://github.com/input-output-hk/cardano-wallet/blob/master/src/Cardano/Wallet/Network/HttpBridge.hs#L160-L163)

Reply:
OK, moved. I tried both

```haskell
getNetworkTip :: ExceptT ErrNetworkUnreachable m (Maybe (Hash "BlockHeader", BlockHeader))
-- and
getNetworkTip :: ExceptT ErrNetworkTip m (Hash "BlockHeader", BlockHeader)
```

The latter seems better.